diff --git a/lib/facter/operatingsystem.rb b/lib/facter/operatingsystem.rb
index 7cb10f5..f8e666e 100644
--- a/lib/facter/operatingsystem.rb
+++ b/lib/facter/operatingsystem.rb
@@ -38,6 +38,8 @@ Facter.add(:operatingsystem) do
        "Ubuntu"
     elsif FileTest.exists?("/etc/debian_version")
       "Debian"
+    elsif FileTest.exists?("/etc/openwrt_release")
+      "OpenWrt"
     elsif FileTest.exists?("/etc/gentoo-release")
       "Gentoo"
     elsif FileTest.exists?("/etc/fedora-release")
diff --git a/lib/facter/operatingsystemrelease.rb b/lib/facter/operatingsystemrelease.rb
index 2daecd6..29d356d 100644
--- a/lib/facter/operatingsystemrelease.rb
+++ b/lib/facter/operatingsystemrelease.rb
@@ -93,6 +93,17 @@ Facter.add(:operatingsystemrelease) do
 end
 
 Facter.add(:operatingsystemrelease) do
+  confine :operatingsystem => %w{OpenWrt}
+  setcode do
+    if release = Facter::Util::Resolution.exec('cat /etc/openwrt_release')
+      if match = /^DISTRIB_RELEASE="(\d+\.\d+)"/.match(release)
+        match[1]
+      end
+    end
+  end
+end
+
+Facter.add(:operatingsystemrelease) do
   confine :operatingsystem => %w{Slackware}
   setcode do
     if release = Facter::Util::FileRead.read('/etc/slackware-version')
